website/docs/security/security-hardening.md

```diff
@@ -36,3 +36,28 @@ To prevent any user from creating/editing CAPTCHA stages block API requests to t
 -   `/api/v3/managed/blueprints*`
 
 With these restrictions in place, CAPTCHA stages can only be edited using [Blueprints on the file system](../customize/blueprints/index.md#storage---file).
+
+### Content Security Policy (CSP)
+
+:::caution
+Setting up CSP incorrectly may result in the client not loading necessary third-party code.
+:::
+
+Content Security Policy (CSP) is a security standard that mitigates the risk of content injection vulnerabilities. authentik doesn't currently support CSP natively, so setting it up depends on your installation. We recommend using a [reverse proxy](../install-config/reverse-proxy.md) to set a CSP header.
+
+authentik will require at least the following origins:
```

Comment:
```suggestion
authentik requires at least the following origins:
```

Reply:
Should we explain more about what an origin is? Where to define it? Do we have any links about it to link to?

Reply:
Offline resolution: we should use the word used by the CSP spec: "location"